realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -57,41 +68,31 @@ public Long get() {
         @Override
         public void increment(long inc) {
             if (value == null) {
-                throw new NullPointerException("Attempt to increment a null valued MutableRealmInteger");
+                throw new IllegalArgumentException("Attempt to increment a null valued MutableRealmInteger");
             }
             value = Long.valueOf(value + inc);
         }
 
         @Override
         public void decrement(long dec) {
-            if (value == null) {
-                throw new NullPointerException("Attempt to decrement a null valued MutableRealmInteger");
-            }
-            value = Long.valueOf(value - dec);
-        }
-
-        @Override
-        public boolean isManaged() {
-            return false;
+            increment(-dec);
         }
 
         @Override
-        public boolean isValid() {
-            return true;
+        public boolean isNull() {
+            return get() == null;
         }
     }
 
 
     /**
      * Managed Implementation.
      */
-    // FIXME MutableRealmIntegers: wire up to native increment method
     static class Managed extends MutableRealmInteger {
         private final ProxyState<?> proxyState;
         private final BaseRealm realm;
         private final Row row;
         private final long columnIndex;
```

Comment:
`columnIndex` can be updated by `SharedRealm.refresh()` because of additive schema change from ROS.
We always need to refer `FooColumnInfo` in proxy class if we use columnIndex cache. Or proxy class must update this `columnIndex` field.